craft_parts/errors.py

```diff
@@ -837,19 +841,22 @@ def __init__(
 
     @classmethod
     def from_validation_error(
-        cls, *, error_list: list["ErrorDict"]
-    ) -> "FilesystemMountError":
+        cls, *, error_list: list[ErrorDetails]
+    ) -> FilesystemMountError:
         """Create a FilesystemMountError from a pydantic error list.
 
         :param error_list: A list of dictionaries containing pydantic error definitions.
         """
         formatted_errors: list[str] = []
 
         for error in error_list:
-            loc = error.get("loc")
-            msg = error.get("msg")
+            try:
+                loc = error["loc"]
+                msg = error["msg"]
+            except KeyError:
+                continue
```

Comment:
Is a loc value of `None` allowed here? Because if the key exists and the value is `None` the conditional below won't catch it. If it's the same case as the previous similar block we could use `get` instead.